src/model/document.ts

```diff
@@ -138,6 +138,10 @@ export class Document implements Feature, Queryable {
     this.warnings = Array.from(base.warnings);
   }
 
+  static isDocument(value: any): value is Document {
+    return typeof value === 'object' && !!value.constructor.isDocument;
```

Comment:
This throws if given `null`. `typeof null` is 'object' D: :D D: :D

How about just:

```
return value && value.constructor && value.constructor.isDocument;
```